core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -356,9 +381,6 @@ void CudaSpace::impl_deallocate(
       KOKKOS_IMPL_CUDA_SAFE_CALL(cudaSetDevice(m_device));
       KOKKOS_IMPL_CUDA_SAFE_CALL(cudaFree(arg_alloc_ptr));
     }
-#else
-    KOKKOS_IMPL_CUDA_SAFE_CALL(cudaSetDevice(m_device));
-    KOKKOS_IMPL_CUDA_SAFE_CALL(cudaFree(arg_alloc_ptr));
```

Comment:
What happened to this branch?

Reply:
isn't that the stuff in 367-369?

Reply:
This is the fallback branch for when `KOKKOS_ENABLE_IMPL_CUDA_MALLOC_ASYNC` is false or we have a Cuda version less than 11.2. With the changes here we don't free the memory in this case anymore AFAICT.